sinfonier-web/app/controllers/Topologies.java

```diff
@@ -203,8 +207,74 @@ public static void search(String status, String query, Date updated, int page) t
   public static void log(@Required String id) {
     try {
       Codes code200 = Codes.CODE_200;
+      Gson gson = new Gson();
       JsonObject data = new JsonObject();
-      data.addProperty("msg", client.getTopologyLog(id));
+      String logDataKey = id+"_logdata";
+      String startParam = request.params.get("start");
```

Comment:
Yes. for example `public static void log(@Required String id, Integer start) ` if you don't pass the param it will be null, the only important thing is that the type **must** be an **object**. You can take a look in the _index_ method.